packages/@cosmos-building-blocks/service/src/redis.ts

```diff
@@ -0,0 +1,104 @@
+import { Construct, IConstruct } from '@aws-cdk/core';
+import { CfnSubnetGroup, CfnReplicationGroup, CfnReplicationGroupProps } from '@aws-cdk/aws-elasticache';
+import { SecurityGroup, Port, IVpc, Connections } from '@aws-cdk/aws-ec2';
+
+export interface IRedis extends IConstruct {
+  /**
+   * Subnet Group created by construct using provided Subnets
+   */
+  readonly cacheSubnetGroup: CfnSubnetGroup;
+  /**
+   * Security Group assigned to Redis Cluster
+   */
+  readonly redisSecurityGroup: SecurityGroup;
+  /**
+   * Actual Redis Cluster
+   */
+  readonly redisReplicationGroup: CfnReplicationGroup;
+}
+
+export interface RedisProps {
+  /**
+   * VPC to launch Redis cluster in
+   */
+  vpc: IVpc;
+  /**
+   * Allowed ingress connections to Redis SecurityGroup
+   *
+   * @default - Allowed from anywhere.
+   */
+  connectionsAllowedFrom?: Connections;
+  /**
+   * List of Subnets to launch Redis cluster in
+   */
+  subnetIds: string[];
```

Comment:
can we change this to a subnet selection array as thats the cdk standard ?

Reply:
move up one place

Reply:
Done